### PR TITLE
[JCLOUDS-1135] improve performance of AzureComputeServiceAdapter.getNode

### PR DESCRIPTION
**Pull request by Ladicek**
The implementation is now O(1) in the number of existing cloud services
instead of O(N). This is achieved by assuming that deployment name
is the same as cloud service name, which is true for all nodes created
by JClouds Azure provider.

### CODE REVIEW

**Comment by nacx** [comment]
Thanks @Ladicek!
To make this backwards compatible and not lose the ability to manage nodes not created by jclouds (this is something we should be able to do), could you default to the old behaviour if the deployment name is not the expected one? This way we would have good performance for nodes created by jclouds, and keep the old behaviour for the rest. WDYT?

**Comment by Ladicek** [comment]
@nacx I pushed a change where the new behavior has to be enabled explicitly. Is that what you had in mind?

**Comment by nacx** [comment]
Not exactly, instead of introducing a new configuration property I was thinking about something like:
```java
final Deployment deployment = api.getDeploymentApiForService(id).get(id);
if (deployment == null || deployment.roleInstanceList().isEmpty()) {
   // Instead of returning null, call the old logic that lists all cloud services
}
```
Does this make sense?

**Comment by Ladicek** [comment]
Oh, OK, I'll take a stab at this.

**Comment by Ladicek** [comment]
OK, I pushed another change.

File: azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java [hunk 1]
```diff
@@ -267,6 +267,19 @@ public boolean apply(final OSImage input) {
 
    @Override
    public Deployment getNode(final String id) {
+      final Deployment deployment = api.getDeploymentApiForService(id).get(id);
+      if (deployment != null && !deployment.roleInstanceList().isEmpty()) {
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java]: This conditional is the same one than the used below. Refactor this a bit to reuse the code.

**Comment by nacx** [comment]
The new approach lgtm. Thanks! @andreaturli do you want to have a look?

**Comment by Ladicek** [comment]
I originally kept that small duplicity in place to make the diff smaller, but I factored the common bit out as you wish :-)

**Comment by nacx** [comment]
Pushed to master as [7dd02a01](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/7dd02a01). Thanks @Ladicek! :)

**Pull request closed without merging**